main.go

```diff
@@ -89,9 +91,22 @@ func run(l *slog.Logger) error {
 	// create the namespace lister
 	nsl := NewNamespaceListerForSubject(accessCache)
 
+	// create metrics tracker
+	reg := NewDefaultRegistry()
+	metrics := NewMetrics(reg)
+
+	// serve metrics
+	metricsServer := NewMetricsServerBuilder().
+		WithAddress(metricsListenAddress).
+		WithTLS(enableTLS).
+		WithTLSConfig(loadTLSCert(l, tlsCertificatePath, tlsCertificateKeyPath)).
+		WithRegistry(reg).
+		Build()
```

Comment:
I'd suggest to provide the required parameters in the constructor and leave the `With*` approach for optional ones